src/main/java/ua/orlov/springcoregym/dao/impl/training/TrainingDaoImpl.java

```diff
@@ -0,0 +1,108 @@
+package ua.orlov.springcoregym.dao.impl.training;
+
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.TypedQuery;
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.stereotype.Repository;
+import ua.orlov.springcoregym.dao.AbstractDao;
+import ua.orlov.springcoregym.dto.training.TraineeTrainingsRequest;
+import ua.orlov.springcoregym.dto.training.TrainerTrainingRequest;
+import ua.orlov.springcoregym.model.training.Training;
+
+import java.util.List;
+
+
+@Log4j2
+@Repository
+public class TrainingDaoImpl extends AbstractDao<Training, Long> implements TrainingDao {
+
+    private static String GET_TRAININGS_BY_TRAINEE_TRAININGS_REQUEST;
```

Comment:
Why is this field static and initialized in Consturtor?
And why name in upper case? we don't use here final modifier